### PR TITLE
Valid signatures

### PR DESCRIPTION
**Pull request by tschmidtb51**
- addresses parts of https://github.com/oasis-tcs/csaf/issues/678
- add FAQ on signing
- provide tool guidance

### CODE REVIEW

File: csaf_2.0/guidance/faq.md [hunk 1]
```diff
@@ -48,6 +48,15 @@ Starting with version CSAF 2.1, [CSAF will support TLP v2](https://github.com/oa
 
 CSAF lister and CSAF aggregator choose on their own which producing parties they add to their lists. Please reach out to the CSAF lister or CSAF aggregator in question. Their contact details are available in the metadata of the list.
 
+### How long should signatures of CSAF documents be valid?
+
+At any given point in time, signatures MUST be valid for 30 more days and SHOULD be valid for at least 90 more days. When signing CSAF documents, the signing party SHOULD comply with or exceed current best practices and guidance on key length.
+Tools SHOULD treat the violation of the rules given in the first sentence as:
+
+* warning if the signature is only valid for 90 days or less at the time of the check,
```
Comment (sthagen) [inline csaf_2.0/guidance/faq.md:56]: maybe `check` &rightarrow; `verification` in all three statements (list items)?

**Review by sthagen — APPROVED** [review]
LGTM when replacing `check` with `verification`.

File: csaf_2.0/guidance/faq.md [hunk 2]
```diff
@@ -48,6 +48,15 @@ Starting with version CSAF 2.1, [CSAF will support TLP v2](https://github.com/oa
 
 CSAF lister and CSAF aggregator choose on their own which producing parties they add to their lists. Please reach out to the CSAF lister or CSAF aggregator in question. Their contact details are available in the metadata of the list.
 
+### How long should signatures of CSAF documents be valid?
+
+At any given point in time, signatures MUST be valid for 30 more days and SHOULD be valid for at least 90 more days. When signing CSAF documents, the signing party SHOULD comply with or exceed current best practices and guidance on key length.
```
Comment (santosomar) [inline csaf_2.0/guidance/faq.md:53]: I humbly suggest to change to something similar to this for readability:
"At all times, signatures MUST remain valid for a minimum of 30 days and ideally for at least 90 days. When executing CSAF document signatures, the signing party should adhere to or surpass the prevailing best practices and recommendations regarding key length."

**Review by santosomar — CHANGES_REQUESTED** [review]

File: csaf_2.0/guidance/faq.md (hunk 1 above)
Comment (tschmidtb51) [reply csaf_2.0/guidance/faq.md]: Done.

File: csaf_2.0/guidance/faq.md (hunk 2 above)
Comment (tschmidtb51) [reply csaf_2.0/guidance/faq.md]: @santosomar I took your suggestion. However, I'm still wondering about the part "ideally for at least 90 days" as a reader might misunderstand it in a way that you should change signatures every 90 days.

**Review by sthagen — APPROVED** [review]
Nice.

**Comment by santosomar** [comment]
A motion will be put in place to approve the contribution as stated after today's TC meeting.

**Review by santosomar — APPROVED** [review]

**Comment by santosomar** [comment]
This motion has automatically passed on 2024-05-01 23:00 UTC.

https://groups.oasis-open.org/discussion/motion-for-724

**Pull request merged**